core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -103,6 +104,81 @@ private[spark] class AppStatusListener(
     }
   }
 
+  // visible for tests
+  private[spark] def recoverLiveEntities(): Unit = {
+    if (!live) {
+      kvstore.view(classOf[JobDataWrapper])
+        .asScala.filter(_.info.status == JobExecutionStatus.RUNNING)
```

Comment:
Safer condition would be either `UNKNOWN` or `RUNNING`, but I guess it shouldn't be UNKNOWN so seems OK.

Reply:
Yeah. I didn't see any place we set Job status to `UNKNOWN`.